mapclientplugins/geometricfitstep/view/geometricfitwidget.py

```diff
@@ -528,6 +545,76 @@ def _configProjectionCentreGroupsClicked(self):
                 config.run()
                 self._refreshStepItem(config)
                 self._refreshGraphics()
+    
+    def _configSettingGroupChanged(self, index):
+        """
+        Callback for change in setting group field chooser widget.
+        """
+        self._updateDataProportion()
+
+    def _updateDataProportion(self):
+        realFormat = "{:.4g}"
+        lineEditDisable = True
+        checkBoxDisable = True
+        checkBoxTristate = False
+        dataProportionStr = "1.0"
+        checkBoxState = QtCore.Qt.Unchecked
+        config = self._getConfig()
+        group = self._ui.configSettingGroup_fieldChooser.getField()
+        if group:
+            checkBoxDisable = False
+            proportion, isLocallySet = config.getGroupDataProportion(group.getName())
+            #print("group:", group.getName(), "proportion", proportion, isLocallySet)
+            if proportion:
+                dataProportionStr = realFormat.format(proportion)
+                if isLocallySet:
```

Comment:
The locally set case needs to set checkBoxTristate=True if there is a previous value. You can technically find this out by seeing if there is a previous config step which has set the value. I suggest getting that workaround going, but we can talk about making scaffoldfitter's API more convenient for this.